src/storage/observer_worker.rs

```diff
@@ -85,6 +107,37 @@ where
                         .await;
                 }
             }
+            OperationType::DeferredDumpBlobIndexes => self.deffer_blob_indexes_dump().await?,
+        }
+        Ok(())
+    }
+
+    async fn deffer_blob_indexes_dump(&self) -> Result<()> {
+        let mut deferred = self.deferred.write().await;
+        if let Some(ref mut deferred) = *deferred {
+            deferred.update_last_time();
+        } else {
+            let sender = self.loopback_sender.clone();
+            let data = DeferredEventData::new();
+            *deferred = Some(data);
+            let min = self.deferred_min_duration.clone();
+            let max = self.deferred_max_duration.clone();
+            let deferred = self.deferred.clone();

```

Comment:
`deferred` redefined here, so it is not easy to find out what `deferred` is moved into new thread. I suggest to give a different name here